app/modules/enduser_setup.c

```diff
@@ -625,6 +626,228 @@ static void do_station_cfg (task_param_t param, uint8_t prio)
   luaM_free(lua_getstate(), cnf);
 }
 
+enum HTTP_FIELD_PARSER {
+  STATE_FIELD = 1,
+  STATE_DATA = 2,
+  STATE_FIN = 3
+};
+
+// we could detect how many params here (by counting ?), but not sure that is wise
+const int MAX_HTTP_FIELDS = 10;
+
+static int decode_http_queryparam(char *data, unsigned short data_len, char *fields[], char *values[]) {
+  unsigned int counter = 0;
+
+  int lastPos = -1;
+  int state = STATE_FIELD;
+  int pos;
+
+  for(pos = 0; pos < MAX_HTTP_FIELDS; pos ++) {
+    fields[pos] = values[pos] = NULL;
+  }
+
+  for (pos = 0; pos < data_len; pos++) {
+    if (data[pos] == '?' && (pos + 1 < data_len)) {
+      fields[counter] = &data[pos + 1];
+      break;
+    }
+  }
+
+  data[data_len] = 0; // this must be true by definition
+
+  for (pos = 0; pos < data_len && counter < MAX_HTTP_FIELDS && state != STATE_FIN; pos++) {
+    //std::cout << "val" << pos << ":" << data[pos] << std::endl;
+    if (data[pos] == '&' || data[pos] == ';') { // w3c states & or ; separate params
+      if (lastPos != -1) {
+        data[pos] = 0; // terminate the string
+        counter++;
+
+        if (counter < MAX_HTTP_FIELDS) {
+          fields[counter] = NULL;
```

Comment:
There is a loop above that initializes all of these to NULL. Do you need to do it again?